dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -86,7 +88,8 @@ private DefaultFuture(Channel channel, Request request, int timeout) {
      */
     private static void timeoutCheck(DefaultFuture future) {
         TimeoutCheckTask task = new TimeoutCheckTask(future);
-        TIME_OUT_TIMER.newTimeout(task, future.getTimeout(), TimeUnit.MILLISECONDS);
+        Timeout t = TIME_OUT_TIMER.newTimeout(task, future.getTimeout(), TimeUnit.MILLISECONDS);
+        PENDING_TASKS.put(future.getId(), t);
```

Comment:
Can we avoid this `PENDING_TASKS` instance? I think `DefaultFuture` itself hosts the status of its timeouttask or `TIME_OUT_TIMER ` hosts the status of all timeout tasks can be better.

I need to merge this PR first to do some pressure test, please help to improve if possible.